waddrmgr/db.go

```diff
@@ -292,6 +376,64 @@ func fetchWatchingOnly(tx walletdb.Tx) (bool, error) {
 	return buf[0] != 0, nil
 }
 
+// fetchNumAllAddresses loads the metadata - number of all addrs - from the database.
+func fetchNumAllAddresses(tx walletdb.Tx) (uint32, error) {
+	bucket := tx.RootBucket().Bucket(metaBucketName)
+
+	val := bucket.Get(numAllAddrsName)
+	// Return 0 if the metadata is uninitialized
+	if val == nil {
+		return 0, nil
+	}
+	if len(val) != 4 {
+		str := fmt.Sprintf("malformed metadata '%s' stored in database",
+			numAllAddrsName)
+		return 0, managerError(ErrDatabase, str, nil)
+	}
+	n := binary.LittleEndian.Uint32(val[0:4])
+	return n, nil
+}
+
+// fetchNumAccountAddrs loads the metadata - number of account addrs - from the database.
+func fetchNumAccountAddrs(tx walletdb.Tx, account uint32) (uint32, error) {
+	bucket := tx.RootBucket().Bucket(metaBucketName).Bucket(uint32ToBytes(account))
+	// Return 0 if bucket is uninitialized
+	if bucket == nil {
+		return 0, nil
+	}
+
+	val := bucket.Get(numAcctAddrsName)
+	// Return 0 if the metadata is uninitialized
+	if val == nil {
```

Comment:
Same deal here. This data should be initialized when the bucket is created and be an error if it's missing here.